docs/contribute/addnode.qmd

```diff
@@ -0,0 +1,170 @@
+---
+title: "How to add a new node type"
+---
+
+Several parts of the code have to be made aware of the new node type. In the rest of this page we shall call our new node type `NewNodeType`.
+
+# The Julia core
+
+## Parameters
+
+The parameters object (defined in `solve.jl`) passed to the ODE solver must be made aware of the new node type. Therefore define a struct in `solve.jl` which holds the data for each node of the new node type:
+
+```julia
+struct NewNodeType
+    node_id::Vector{Int}
+    # Other fields
+end
+```
+
+This struct should at least contain a vector of all node IDs that are of the new type in a given model. Now you can:
+
+- add `new_node_type::NewNodeType` to the Parameters object;
+- add `new_node_type = NewNodeType(db,config)` to the function `Parameters` in create.jl and add new_node_type add the proper location in the `Parameters` constructor call.
+
+## Reading from configuration
+
+To define a schema for the new node type, add the following to `validation.jl`:
+
+```julia
+@version NewNodeTypeV1 begin
+	node_id::Int
+	# Other fields
+end
+```
+
+Now we define the function that is called in the second bullet above, in `create.jl`:
+
+```julia
+function NewNodeType(db::DB, config::Config)::NewNodeType
+	static = load_structvector(db, config, NewNodeTypeV1)
+
+	# Unpack the fields of static as inputs for the NewNodeType constructor
+	return NewNodeType(static.node_id, static.some_property)
+end
+```
+
+## Node behaviour
```

Comment:
Nit, but I think generally the docs are in American English, e.g. we so far only use "behavior".